mlir/test/python/dialects/irdl.py

```diff
@@ -0,0 +1,66 @@
+# RUN: %PYTHON %s 2>&1 | FileCheck %s
+
+from mlir.ir import *
+from mlir.dialects import irdl
+import sys
+
+
+def run(f):
+    print("\nTEST:", f.__name__, file=sys.stderr)
+    f()
+
+
+# CHECK: TEST: testIRDL
+@run
+def testIRDL():
+    with Context() as ctx, Location.unknown():
+        module = Module.create()
+        with InsertionPoint(module.body):
+            dialect = irdl.DialectOp("irdl_test")
+            with InsertionPoint(dialect.body):
+                op = irdl.OperationOp("test_op")
+                with InsertionPoint(op.body):
+                    f32 = irdl.is_(TypeAttr.get(F32Type.get()))
+                    irdl.operands_([f32], ["input"], [irdl.Variadicity.single])
+                type1 = irdl.TypeOp("type1")
```

Comment:
Hmm - if you replace this line with `type1 = irdl.type("type1")` (or `type1 = irdl.type_("type1")`), does `type1.body` still work in the next line?

If not, could you add the snake_case wrappers for the (extended) ops defs you have in your `irdl.py` file? For all four of them, please.